test/lib/index.test.ts

```diff
@@ -0,0 +1,56 @@
+import { test } from 'tap';
+import * as path from 'path';
+
+import { readFixture } from '../utils';
+import { buildDepTree, LockfileType } from '../../lib';
+
+test('Default manifest name is used when manifest is missing', async (t) => {
+  // Arrange
+  const manifestFileContent: string = await readFixture(
+    'package-json-without-name/package.json',
+  );
+  const lockFileContent: string = await readFixture(
+    'package-json-without-name/package-lock.json',
+  );
+
+  // Act
+  const result = await buildDepTree(
+    manifestFileContent,
+    lockFileContent,
+    false,
+    LockfileType.npm,
+    true,
+  );
+
+  // Assert
+  t.equal(result.name, 'package.json', 'resolved to correct name');
+});
+
+test('Default manifest name is resolved when path is absolute', async (t) => {
+  // Arrange
+  const manifestFileContent: string = await readFixture(
+    'package-json-without-name/package.json',
+  );
+  const lockFileContent: string = await readFixture(
+    'package-json-without-name/package-lock.json',
+  );
+  const absolutePath = path.resolve(
+    __dirname,
+    'lib',
+    'fixtures',
+    'package.json',
```

Comment:
can you change this to something else to show that this absolutePath is in fact being used properly. (currently both tests have the exact same assert, which makes it hard to see what the difference between both calls is)

Reply:
The difference is in incoming parameters. That output should be the same. That is why I have same asserts :)